xwiki-pro-macros-test/xwiki-pro-macros-test-pageobjects/src/main/java/com/xwiki/pro/test/po/generic/TeamMacroPage.java

```diff
@@ -37,11 +38,82 @@ public class TeamMacroPage extends ViewPage
     @FindBy(css = ".xwikiteam")
     private List<WebElement> teamMacros;
 
-    public int getTeamMacrosCount() {
+    public int getTeamMacrosCount()
+    {
         return teamMacros.size();
     }
 
-    public List<WebElement> getTeamMacroUsers(int i) {
+    public List<WebElement> getTeamMacroUsers(int i)
+    {
         return teamMacros.get(i).findElements(By.className("xwikiteam-user"));
     }
+
+    public String getUserTitle(int macroIndex, String dataUsername)
+    {
+        return getUserByUsername(macroIndex, dataUsername).getAttribute("title");
+    }
+
+    public String getProfileLink(int macroIndex, String dataUsername)
+    {
+        return getUserByUsername(macroIndex, dataUsername)
+            .findElement(By.tagName("a")).getAttribute("href");
+    }
+
+    public String getAvatarInitials(int macroIndex, String dataUsername)
+    {
+        return getUserByUsername(macroIndex, dataUsername)
+            .findElement(By.className("xwikiteam-avatar-initials-letters")).getText();
+    }
+
+    public String getAvatarBackgroundColor(int macroIndex, String dataUsername)
+    {
+        return getUserByUsername(macroIndex, dataUsername)
+            .findElement(By.className("xwikiteam-avatar")).getCssValue("background-color");
+    }
+
+    public String getAvatarFontColor(int macroIndex, String dataUsername)
+    {
+        return getUserByUsername(macroIndex, dataUsername)
+            .findElement(By.className("xwikiteam-avatar-initials-letters")).getCssValue("color");
+    }
+
+    public String getAvatarSize(int macroIndex, String dataUsername)
+    {
+        return getUserByUsername(macroIndex, dataUsername)
+            .findElement(By.className("xwikiteam-avatar")).getCssValue("height");
+    }
+
+    public String getAvatarBorderRadius(int macroIndex, String dataUsername)
+    {
+        return getUserByUsername(macroIndex, dataUsername)
+            .findElement(By.className("xwikiteam-avatar")).getCssValue("border-radius");
+    }
+
+    private WebElement getUserByUsername(int macroIndex, String dataUsername)
+    {
+        return teamMacros.get(macroIndex)
+            .findElements(By.className("xwikiteam-user"))
+            .stream()
+            .filter(user -> dataUsername.equals(user.getAttribute("data-username")))
+            .findFirst()
+            .orElseThrow(() -> new NoSuchElementException("User not found: " + dataUsername));
+    }
+
+    public boolean isUsernameHidden(int macroIndex)
+    {
+        WebElement container = teamMacros.get(macroIndex);
+        String classAttr = container.getAttribute("class");
+        return classAttr.contains("usernames-hidden");
```

Comment:
Maybe you could check if the username is displayed instead? So you verify that `usernames-hidden` does what it should do